homeassistant/components/sensor/whois.py

```diff
@@ -0,0 +1,135 @@
+"""
+Get WHOIS information for a given host.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.whois/
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_NAME
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['pythonwhois==2.4.3']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_DOMAIN = 'domain'
+
+DEFAULT_NAME = 'Whois'
+
+ATTR_NAME_SERVERS = 'name_servers'
+ATTR_REGISTRAR = 'registrar'
+ATTR_UPDATED = 'updated'
+ATTR_EXPIRES = 'expires'
+
+SCAN_INTERVAL = timedelta(hours=24)  # WHOIS info is very slow moving
+# We also want to prevent DOS / TOS breaking; One request per domain
+# every 24 hours shouldn't count as "high volume"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_DOMAIN): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string
+})
+
+
+def setup_platform(hass, config, add_devices, disovery_info=None):
+    """Set up the WHOIS sensor."""
+    from pythonwhois import get_whois
+    from pythonwhois.shared import WhoisException
+
+    domain = config.get(CONF_DOMAIN)
+    name = config.get(CONF_NAME)
+
+    try:
+        if 'expiration_date' in get_whois(domain, normalized=True):
```

Comment:
Especially for European domains is the `expiration_date` not always present. So the lookup is working but we will not get the data.

Reply:
Not much we can do here. Perhaps re-phrasing the warning.